{
  message: "Failed to Connect Database Could not connect to any servers in your MongoDB Atlas cluster. One common reason is that you're trying to access the database from an IP that isn't whitelisted. Make sure your current IP address is on your Atlas cluster's IP whitelist: https://www.mongodb.com/docs/atlas/security-whitelist/",
  reason: TopologyDescription {
    type: 'ReplicaSetNoPrimary',
    servers: Map(3) {
      'ac-teig1qz-shard-00-00.c4kfkmr.mongodb.net:27017' => ServerDescription {
        address: 'ac-teig1qz-shard-00-00.c4kfkmr.mongodb.net:27017',
        type: 'Unknown',
        hosts: [],
        passives: [],
        arbiters: [],
        tags: {},
        minWireVersion: 0,
        maxWireVersion: 0,
        roundTripTime: -1,
        lastUpdateTime: 1203649858,
        lastWriteDate: 0,
        error: MongoNetworkError: connect ETIMEDOUT 3.7.19.65:27017
            at connectionFailureError (D:\web-development\level-2\university-management-auth-service\node_modules\mongodb\src\cmap\connect.ts:491:14)
            at TLSSocket.<anonymous> (D:\web-development\level-2\university-management-auth-service\node_modules\mongodb\src\cmap\connect.ts:392:16)
            at Object.onceWrapper (node:events:628:26)
            at TLSSocket.emit (node:events:513:28)
            at emitErrorNT (node:internal/streams/destroy:151:8)
            at emitErrorCloseNT (node:internal/streams/destroy:116:3)
            at processTicksAndRejections (node:internal/process/task_queues:82:21) {
          cause: Error: connect ETIMEDOUT 3.7.19.65:27017
              at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1494:16) {
            errno: -4039,
            code: 'ETIMEDOUT',
            syscall: 'connect',
            address: '3.7.19.65',
            port: 27017
          },
          [Symbol(errorLabels)]: Set(1) { 'ResetPool' }
        },
        topologyVersion: null,
        setName: null,
        setVersion: null,
        electionId: null,
        logicalSessionTimeoutMinutes: null,
        primary: null,
        me: null,
        '$clusterTime': null
      },
      'ac-teig1qz-shard-00-01.c4kfkmr.mongodb.net:27017' => ServerDescription {
        address: 'ac-teig1qz-shard-00-01.c4kfkmr.mongodb.net:27017',
        type: 'Unknown',
        hosts: [],
        passives: [],
        arbiters: [],
        tags: {},
        minWireVersion: 0,
        maxWireVersion: 0,
        roundTripTime: -1,
        lastUpdateTime: 1203649856,
        lastWriteDate: 0,
        error: MongoNetworkError: connect ETIMEDOUT 3.6.181.176:27017
            at connectionFailureError (D:\web-development\level-2\university-management-auth-service\node_modules\mongodb\src\cmap\connect.ts:491:14)
            at TLSSocket.<anonymous> (D:\web-development\level-2\university-management-auth-service\node_modules\mongodb\src\cmap\connect.ts:392:16)
            at Object.onceWrapper (node:events:628:26)
            at TLSSocket.emit (node:events:513:28)
            at emitErrorNT (node:internal/streams/destroy:151:8)
            at emitErrorCloseNT (node:internal/streams/destroy:116:3)
            at processTicksAndRejections (node:internal/process/task_queues:82:21) {
          cause: Error: connect ETIMEDOUT 3.6.181.176:27017
              at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1494:16) {
            errno: -4039,
            code: 'ETIMEDOUT',
            syscall: 'connect',
            address: '3.6.181.176',
            port: 27017
          },
          [Symbol(errorLabels)]: Set(1) { 'ResetPool' }
        },
        topologyVersion: null,
        setName: null,
        setVersion: null,
        electionId: null,
        logicalSessionTimeoutMinutes: null,
        primary: null,
        me: null,
        '$clusterTime': null
      },
      'ac-teig1qz-shard-00-02.c4kfkmr.mongodb.net:27017' => ServerDescription {
        address: 'ac-teig1qz-shard-00-02.c4kfkmr.mongodb.net:27017',
        type: 'Unknown',
        hosts: [],
        passives: [],
        arbiters: [],
        tags: {},
        minWireVersion: 0,
        maxWireVersion: 0,
        roundTripTime: -1,
        lastUpdateTime: 1203649859,
        lastWriteDate: 0,
        error: MongoNetworkError: connect ETIMEDOUT 3.6.213.55:27017
            at connectionFailureError (D:\web-development\level-2\university-management-auth-service\node_modules\mongodb\src\cmap\connect.ts:491:14)
            at TLSSocket.<anonymous> (D:\web-development\level-2\university-management-auth-service\node_modules\mongodb\src\cmap\connect.ts:392:16)
            at Object.onceWrapper (node:events:628:26)
            at TLSSocket.emit (node:events:513:28)
            at emitErrorNT (node:internal/streams/destroy:151:8)
            at emitErrorCloseNT (node:internal/streams/destroy:116:3)
            at processTicksAndRejections (node:internal/process/task_queues:82:21) {
          cause: Error: connect ETIMEDOUT 3.6.213.55:27017
              at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1494:16) {
            errno: -4039,
            code: 'ETIMEDOUT',
            syscall: 'connect',
            address: '3.6.213.55',
            port: 27017
          },
          [Symbol(errorLabels)]: Set(1) { 'ResetPool' }
        },
        topologyVersion: null,
        setName: null,
        setVersion: null,
        electionId: null,
        logicalSessionTimeoutMinutes: null,
        primary: null,
        me: null,
        '$clusterTime': null
      }
    },
    stale: false,
    compatible: true,
    heartbeatFrequencyMS: 10000,
    localThresholdMS: 15,
    setName: 'atlas-cpg0j9-shard-0',
    maxElectionId: null,
    maxSetVersion: null,
    commonWireVersion: 0,
    logicalSessionTimeoutMinutes: null
  },
  code: undefined,
  level: 'error',
  stack: "MongooseServerSelectionError: Could not connect to any servers in your MongoDB Atlas cluster. One common reason is that you're trying to access the database from an IP that isn't whitelisted. Make sure your current IP address is on your Atlas cluster's IP whitelist: https://www.mongodb.com/docs/atlas/security-whitelist/\n" +
    '    at _handleConnectionErrors (D:\\web-development\\level-2\\university-management-auth-service\\node_modules\\mongoose\\lib\\connection.js:792:11)\n' +
    '    at NativeConnection.openUri (D:\\web-development\\level-2\\university-management-auth-service\\node_modules\\mongoose\\lib\\connection.js:767:11)\n' +
    '    at processTicksAndRejections (node:internal/process/task_queues:95:5)',
  label: 'right now!',
  timestamp: '2023-06-08T08:16:11.986Z'
}
